lib/rules/prefer-const.js

```diff
@@ -203,6 +205,128 @@ function groupByDestructuring(variables, ignoreReadBeforeAssign) {
 }
 
 /**
+ * Returns a list of nodes
+ * that have have the type Identifier.
+ * This will search for nested Identifiers.
+ *
+ * @param {ASTNode} node - node to search for children and nested identifiers.
+ * @returns {ASTNode[]} list Identifier nodes.
+ */
+function getIdentifiersInArrayDestructureGroup(node) {
+    const identifiers = [];
+
+    if (node.elements) {
+        const numberOfElements = node.elements.length;
+
+        for (let i = 0; i < numberOfElements; i++) {
+            if (!node.elements[i].elements && node.elements[i].type === "Identifier") {
```

Comment:
I'm a bit confused about this line -- is the `!node.elements[i].elements` part redundant? It seems like an `Identifier` node would never have an `elements` property.